### ISSUE TITLE
MySQL Connector/J 8.X Above return Unknown system variable 'query_cache_size'

### ISSUE THREAD

**Issue by zscxst** [apache/shardingsphere#5674]
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response **more than 7 days** and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
4.1.0

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy

### Expected behavior
MySQL Server is 8.0.20
MySQL Connector/J 8.X Above version connection success

### Actual behavior
java.sql.SQLException: Unknown system variable 'query_cache_size'

### Reason analyze (If you can)
[https://github.com/apache/shardingsphere/blob/master/shardingsphere-db-protocol/shardingsphere-db-protocol-mysql/src/main/java/org/apache/shardingsphere/db/protocol/mysql/constant/MySQLServerInfo.java](url) file constant SERVER_VERSION.This constant is returned by the handshake protocol. In java client jar file com.mysql.cj.NativeSession will determine the version number of the server.Code is:
if (!versionMeetsMinimum(8, 0, 3)) {
     queryBuf.append(", @@query_cache_size AS query_cache_size");
      queryBuf.append(", @@query_cache_type AS query_cache_type");
}
But mysql server 8 abandon ‘query_cache_size’

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
Mysql server 8.0.20

Java client maven import
```xml
<dependency>
    <groupId>mysql</groupId>
    <artifactId>mysql-connector-java</artifactId>
    <version>8.0.20</version>
</dependency>
```

**Comment by cherrylzhao** [comment]
duplicate with #3097

**Comment by little-liang** [comment]
ask common

**Comment by zscxst** [comment]
> duplicate with #3097

There can be a default version number, but I suggest that you can configure the version number.

**Comment by cherrylzhao** [comment]
fix at #5720
